doc/layer2-forwarding-enhancements/SONiC Layer 2 Forwarding Enhancements HLD.md

```diff
@@ -0,0 +1,444 @@
+
+
+A new  unordered_map containing a mapping between SAI object ID (can be either Port object ID or Bridge Port object ID or VLAN object ID) and Name (port or VLAN alias) will be added which can be looked up in O(1) time. The Port or VLAN alias retrieved from this will be looked up in the Port data structure.
+
+
+	**Object ID to alias mapping.**
+
+
+<img src="images/oidToName.jpg" width="800" height="500">
+
+
+**Changes for FDB flush.**
+When a FDB flush request is received, Orchagent will send a bulk flush request to syncd. Individual FDB delete response from SAI will be used to delete FDB entries in ASIC_DB, STATE_DB and orchagent data structure. The delete response will be received as AGED event to syncd, which will delete the entry in ASIC_DB and notify Orchagent.
+
+
+When a port operational status goes down, portsorch will call fdborch API to trigger FDB flush. Only dynamic FDB will be flushed.
+
+
+When a port is removed from VLAN, portsorch will call fdborch API to trigger FDB flush. Both static and dynamic FDB will be flushed. Also, the static FDB will be added to the temporary DB in orchagent when delete response is received, so that it can programmed back when port is added back to VLAN.
+
+
+When Spanning tree state changes, protocol component within orchagent will call fdborch API to flush FDB by either per Port or per Port per VLAN. Only dynamic FDB will be flushed.
+
+
+	**FDB Flush due to Port operational state down:**
+
+
+<img src="images/portDownFlush.jpg" width="800" height="500">
+
+
+	**FDB Flush due to Port removal from VLAN:**
+
+
+<img src="images/portVlanRemoveFlush.jpg" width="800" height="500">
+
+
+	**FDB Flush due to spanning tree state change:**
+
+
+<img src="images/stpFlush.jpg" width="800" height="500">
+
+
+**Changes for handling MAC move events.**
+Hardware can generate a single MAC move event instead of generating 2 events (del-old mac and add-new mac). Orchagent code changes will be done to replace the existing FDB entry with the new port in STATE_DB. 
+
+
+**Changes for FDB aging time configuration.**
+A new CLI commands will be added for configuring the FDB aging time.
+FDB aging time configuration will be set in the SWITCH table in CONFIG_DB. Vlanmgr will populate it in the APP_DB. Orchagent will get notified and handling will be SwitchOrch. SwitchOrch will call sai_redis API to send request to syncd via ASIC_DB.
+
+
+<img src="images/agingTimeConfig.jpg" width="800" height="500">
+
+
+**Changes for Static FDB configuration.**
+New CLI will be added for configuring Static FDB entry. It will be set in the FDB_TABLE in CONFIG_DB. Vlanmgr will handle the CONIG_DB changes, do necessary validations and then populate in APP_DB. FdbOrch already has all the necessary handling for Static FDB changes from APP_DB.
```

Comment:
I am curious When a static FDB entry is removed from CLI, will SAI notify a SAI_FDB_EVENT_AGED to orchagent?

From images/staticMacConfig.jpg,  even if such an event is present, orchagent will ignore it

Reply:
yes, SAI_FDB_EVENT_AGED will be received and it will be ignored at orchagent because the fdb entry does not exist anymore.